specs/bridge.md

```diff
@@ -1,62 +1,210 @@
-rollup ID, which is the rollup ID that the rollup reads its block data from
-- the rollup enshrines the sequencer bridge account into its consensus,
-authorizing transfers (locks) into this bridge account of a specific asset
-to result in mints of the synthetic token on the rollup
-- users then send transfers to the sequencer bridge account, which result in
-`Deposit` events being included as part of the rollup's block data
-- when the rollup sees a `Deposit`, it mints the corresponding tokens to the
-user's account
-
-This is similar to how existing L1-to-rollup bridges work, where some lock
-event happening on the L1 results in a deposit transaction being derived
-automatically on the rollup.
-
-The rollup-to-sequencer protocol works as follows:
-
-- the bridge account has an associated private key. this could potentially be
-backed by the threshold signature scheme/multisig.
-- a bridge withdrawer is able to sign transaction which move funds out of the
-bridge account to other sequencer accounts.
-- there is a withdrawal contract deployed on the rollup which emits `Withdrawal`
-events when funds are locked in it (and effectively burned, as the contract
-should not be able to transfer its own funds).
-- the bridge withdrawer watches for these `Withdrawal` events. when it sees
-one, it sends a sequencer transaction which transfers funds from the bridge
-account to the account specified in the withdrawal event.
+The protocol is similar to how existing L1-to-rollup bridges work, where a lock
+event on the L1 results in a deposit transaction being derived automatically
+on the rollup.
 
 Since the sequencer can support multiple assets via IBC, the bridging protocol
 also has support for deposits and withdrawals via IBC, as well as support for
 various assets to be bridged to the rollup.
 
-The one-step deposit flow from another IBC chain is as follows:
-
-- a user initiates an ICS20 withdrawal from some IBC chain to Astria.
-- the user sets the destination chain address as the sequencer bridge address
-which corresponds to the rollup they wish to deposit to.
-- the user sets the withdrawal memo to their rollup address.
-- when the ICS20 transfer is executed on Astria, the funds are locked into the
+## High-Level Overview
+
+### Sequencer-to-Rollup
+
+1. The bridge account is initialized on the sequencer with an associated
+rollup ID.
+2. The rollup modifies its consensus to register the sequencer-side bridge account
+for use in its state transition logic.
+3. Users then send sequencer-side transfers to the bridge account (either natively
+or using IBC), which are included in the rollup's block data as `Deposit` events.
+4. Mints to the user's rollup-side account are derived from the `Deposit`s included
+in the rollup block data.
+
+### Rollup-to-Sequencer
+
+1. The bridge withdrawer is registered with the sequencer-side bridge account,
+allowing it to make withdrawals from the bridge account.
```

Comment:
this is a bit confusing, maybe rephrase to "the bridge account has an associated withdrawer address which is authorized to withdraw from the bridge account"